ddtrace/contrib/aiohttp/middlewares.py

```diff
@@ -29,24 +31,32 @@ def attach_context(request):
         service = app[CONFIG_KEY]['service']
         distributed_tracing = app[CONFIG_KEY]['distributed_tracing_enabled']
 
+        context = tracer.context_provider.active()
+
+        # Create a new context based on the propagated information.
+        # Do not fill context with distributed sampling if the tracer is disabled
+        # because the would call the callee to generate references to data which
```

Comment:
OK this is quite fundamental, let me rephrase it then.

Reply:
OK, this really does not apply here so I'm removing the test on `tracer.enabled`. But this would be very important to test for a client. On the server side we don't care as we're free to report whatever we want. But if acting as a client (http client for instance) we should not set the trace ID, parent ID, or distributed sampling attributes if the tracer is disabled. Because then, the server we're calling would think he is responding to a parent trace, but this trace will never be sent to the agent, so it can only result in broken traces.

Reply:
https://github.com/DataDog/dd-trace-py/pull/359/commits/858a4f014218df42975dac22356f5a000fdb355c

Reply:
oh yes that makes sense! will keep track of this for another PR in `requests` and `httplib`